File: scripts/validate_posts_json.js
const fs = require('fs')
const p = 'd:/Cystar Lab/Horse/data/posts.json'
try{
  const s = fs.readFileSync(p,'utf8')
  JSON.parse(s)
  console.log('OK')
}catch(e){
  console.error('ERROR', e.message)
  process.exit(1)
}

File: scripts/validate_posts_json.py
import json
from pathlib import Path

p = Path('d:/Cystar Lab/Horse/data/posts.json')
try:
    data = json.loads(p.read_text(encoding='utf-8'))
    print('OK: parsed', type(data).__name__, 'with', len(data) if hasattr(data,'__len__') else 'N/A', 'items')
except Exception as e:
    print('ERROR', e)
    raise